src/core/qgsmaprenderertask.cpp

```diff
@@ -154,6 +156,38 @@ bool QgsMapRendererTask::run()
       mError = ImageSaveFail;
       return false;
     }
+
+    if ( mSaveWorldFile )
+    {
+      QString content;
+      // note: use 17 places of precision for all numbers output
```

Comment:
I don't like this code being here - I think I should be pulled out to a function which can be reused elsewhere. I'd suggest a static QgsMapSettingsUtils::worldFileContent( const QgsMapSettings &settings). What do you think?